lib/dependencies/WorkerDependency.js

```diff
@@ -74,13 +76,30 @@ WorkerDependency.Template = class WorkerDependencyTemplate extends (
 		runtimeRequirements.add(RuntimeGlobals.baseURI);
 		runtimeRequirements.add(RuntimeGlobals.getChunkScriptFilename);
 
-		source.replace(
-			dep.range[0],
-			dep.range[1] - 1,
-			`/* worker import */ ${RuntimeGlobals.publicPath} + ${
-				RuntimeGlobals.getChunkScriptFilename
-			}(${JSON.stringify(chunk.id)}), ${RuntimeGlobals.baseURI}`
-		);
+		if (dep.crossOriginLoading) {
+			source.replace(
+				dep.range[0],
+				dep.range[1] - 1,
+				`/* worker import */ "data:application/javascript," + ` +
```

Comment:
I choose `data:application/javascript` instead of `Blob` as it's easier to integrate with existing implementation and it's supported by more browsers (and doesn't throw an error in nodejs tests)